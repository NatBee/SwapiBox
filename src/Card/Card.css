.card {
  border: 2px solid #F7F4DB;
  background-color: #CDB578;
  width: 300px;
  justify-content: space-evenly;
  text-align: center;
  margin: 10px;
  font-family: 'Acme', sans-serif; }

.fav-icon {
  background-color: #E6D7B2;
  height: 40px; }

.card .favorite {
  display: none; }

.card div {
  display: flex;
  justify-content: space-between;
  margin: 0;
  flex-wrap: wrap; }

.card-display {
  display: flex;
  font-size: .7em;
  margin: 0;
  padding: 0px 30px; }

.fa-sun {
  background-color: #E6D7B2;
  color: #9A946A;
  font-size: xx-large;
  margin: 6px 0px 0px 262px; }

.fa-sun:hover {
  color: #856B27; }

.title {
  text-align: left;
  width: 100%; }

.value {
  padding: 0px 0px 0px 7px;
  text-align: right; }
