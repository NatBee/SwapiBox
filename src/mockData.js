const mockData = {
  "peopleData": [
    {
    "name": "Beru Whitesun lars",
    "homeworld": undefined,
    "population": undefined,
    "species": undefined,
    "favorite": false
    }
  ],
  "planetData": [
    {
      "name": "Geonosis",
      "climate": "temperate, arid",
      "population": "100000000000",
      "terrain": "rock, desert, mountain, barren",
      "residents": [
        undefined
      ],
      "favorite": false
    }
  ],
  "vehicleData": [ 
    {
      "name": "Sail barge",
      "model": "Modified Luxury Sail Barge",
      "class": undefined,
      "passengers": "500"
    }
   ],
   "movieData": [ 
   {
      "title": "Revenge of the Sith",
      "releaseDate": "2005-05-19",
      "openingCrawl": "War! The Republic is crumbling\r\nunder attacks by the ruthless\r\nSith Lord, Count Dooku.\r\nThere are heroes on both sides.\r\nEvil is everywhere.\r\n\r\nIn a stunning move, the\r\nfiendish droid leader, General\r\nGrievous, has swept into the\r\nRepublic capital and kidnapped\r\nChancellor Palpatine, leader of\r\nthe Galactic Senate.\r\n\r\nAs the Separatist Droid Army\r\nattempts to flee the besieged\r\ncapital with their valuable\r\nhostage, two Jedi Knights lead a\r\ndesperate mission to rescue the\r\ncaptive Chancellor...."
    }
  ]

}

export default mockData;