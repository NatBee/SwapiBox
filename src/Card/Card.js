import React from 'react';
import './Card.css';

const Card = ({data, handleClick}) => {

  const cardDisplayKey = Object.keys(data).map((title, index) => <h3 className="title" key={index}>{title}</h3>);   
  const cardDisplayValues = Object.values(data).map( (value, index) => <h3 className="value" key={index}>{value}</h3>);                          

  return (
    <article className='card'>
      <button onClick={() => handleClick(data)}>
        Favorite
      </button>
      <h3>
        {cardDisplayKey}
      </h3>
      <h3>
        {cardDisplayValues}
      </h3>
    </article>
  );
}

export default Card;