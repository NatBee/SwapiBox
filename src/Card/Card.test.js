/* eslint-disable */ 
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card.js';
import { shallow } from 'enzyme';
import mockData from '../mockData.js';

it('should match the snapshot', () => {
  const mockPeople = mockData.peopleData
  const renderedComponent = shallow(<Card data={mockPeople} />);

  expect(renderedComponent).toMatchSnapshot();
})

it('should have a className of .fav if favorited', () => {

})

it('should have a className of .card if not favorited', () => {
  
})