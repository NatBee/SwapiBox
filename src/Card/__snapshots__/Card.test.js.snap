// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Array [
            Object {
              "homeworld": undefined,
              "name": "Beru Whitesun lars",
              "population": undefined,
              "species": undefined,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          Favorite
</button>,
        <h3>
          <h3
                    className="title"
          >
                    0
          </h3>
</h3>,
        <h3>
          <h3
                    className="value"
          >
                    Object {
                              "homeworld": undefined,
                              "name": "Beru Whitesun lars",
                              "population": undefined,
                              "species": undefined,
                            }
          </h3>
</h3>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="title"
>
              0
</h3>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "0",
              "className": "title",
            },
            "ref": null,
            "rendered": "0",
            "type": "h3",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="value"
>
              Object {
                            "homeworld": undefined,
                            "name": "Beru Whitesun lars",
                            "population": undefined,
                            "species": undefined,
                          }
</h3>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Object {
                "homeworld": undefined,
                "name": "Beru Whitesun lars",
                "population": undefined,
                "species": undefined,
              },
              "className": "value",
            },
            "ref": null,
            "rendered": Object {
              "homeworld": undefined,
              "name": "Beru Whitesun lars",
              "population": undefined,
              "species": undefined,
            },
            "type": "h3",
          },
        ],
        "type": "h3",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            Favorite
</button>,
          <h3>
            <h3
                        className="title"
            >
                        0
            </h3>
</h3>,
          <h3>
            <h3
                        className="value"
            >
                        Object {
                                    "homeworld": undefined,
                                    "name": "Beru Whitesun lars",
                                    "population": undefined,
                                    "species": undefined,
                                  }
            </h3>
</h3>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="title"
>
                0
</h3>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "0",
                "className": "title",
              },
              "ref": null,
              "rendered": "0",
              "type": "h3",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="value"
>
                Object {
                                "homeworld": undefined,
                                "name": "Beru Whitesun lars",
                                "population": undefined,
                                "species": undefined,
                              }
</h3>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Object {
                  "homeworld": undefined,
                  "name": "Beru Whitesun lars",
                  "population": undefined,
                  "species": undefined,
                },
                "className": "value",
              },
              "ref": null,
              "rendered": Object {
                "homeworld": undefined,
                "name": "Beru Whitesun lars",
                "population": undefined,
                "species": undefined,
              },
              "type": "h3",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
